src/schema.rs

```diff
@@ -55,21 +209,124 @@ pub enum Location {
     },
 }
 
+impl Location {
+    pub fn from_json(v: &Value) -> error::Result<Location> {
+        let o = expect_object(v)?;
+        match o.get("parent") {
+            Some(p) => {
+                Ok(
+                    Location::Dependent {
+                        location: expect_string(p)?.to_owned(),
```

Comment:
Keep in mind that a dependent location also contains a property: https://github.com/resin-io/reconfix/blob/rust/docs/schema.markdown#string-property